### ISSUE TITLE
Enable ES6 function name inference

### ISSUE THREAD

**Issue by bterlson** [microsoft/ChakraCore#3407]
There are a number of cases that are currently not working properly, for example:

```js
function foo(a = function () { }) { return a; }
print(foo().name);
```
Expected: 'a'
Actual: ''

This appears to be fixed with the ES6FunctionName flag - we should consider enabling this. Anyone know why this isn't yet enabled?

**Comment by dilijev** [comment]
FWIW I'm not seeing that the `-ES6FunctionName` is not fixing this issue, but that node-ch does the right thing.

```
es -its -h ch-1.6.*,d8,sm,node*,jsc fnameinf.js
## Source
function foo(a = function () { }) { return a; }; print(foo().name); print('end')


┌───────────────┬─────┐
│ ch-1.6.0-args │ end │
│ ch-1.6.0      │     │
├───────────────┼─────┤
│ d8            │ a   │
│ jsc           │ end │
│ sm            │     │
│ node-ch       │     │
│ node          │     │
└───────────────┴─────┘
```

**Comment by bterlson** [comment]
@dilijev perhaps you are right, I noted that -es6experimental fixed it and figured it was because -es6experimental threw ES6FunctionName (which was disabled by default). Perhaps my assumption was wrong :)

**Comment by dilijev** [comment]
Hmm, doesn't repro for me with -es6experimental either. Which version of ChakraCore are you reproing in?

**Comment by farzonl** [comment]
we had a website break 2 years ago. try this flag: ES6FunctionNameFull

**Comment by dilijev** [comment]
Correcting my earlier assertion that it didn't work: I was using a Release build to test. It works with Debug/Test builds.

**Comment by bterlson** [comment]
Another case not enabled in stable is the simple `var x = function () { }; x.name` pattern.

**Comment by rhuanjl** [comment]
This was fixed in master by #6287
